nncf/experimental/torch2/function_hook/extractor.py

```diff
@@ -0,0 +1,222 @@
+# Copyright (c) 2025 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Callable, Dict, List, Optional, Sequence, Tuple
+
+import torch
+from torch import nn
+
+import nncf
+from nncf import nncf_logger
+from nncf.common.graph.graph import NNCFNode
+from nncf.experimental.torch2.function_hook.nncf_graph.layer_attributes import PT2OpLayerAttributes
+from nncf.experimental.torch2.function_hook.wrapper import get_hook_storage
+from nncf.torch.graph import operator_metatypes as om
+from nncf.torch.graph.graph import PTNNCFGraph
+from nncf.torch.model_graph_manager import get_const_data
+from nncf.torch.model_graph_manager import get_const_data_on_port
+from nncf.torch.model_graph_manager import get_const_node
+
+CONV_METATYPES = (
+    om.PTConv1dMetatype,
+    om.PTConv2dMetatype,
+    om.PTConv3dMetatype,
+    om.PTDepthwiseConv1dSubtype,
+    om.PTDepthwiseConv2dSubtype,
+    om.PTDepthwiseConv3dSubtype,
+)
+
+
+class ExtractedFunc(nn.Module):
+    """
+    Module to execute function with kwargs.
+    Support function only with one input.
+
+    :param fn: Function to execute.
+    :param kwargs: Function arguments.
+    """
+
+    def __init__(self, fn: Callable[..., torch.Tensor], kwargs: Dict[str, Any]) -> None:
+        super().__init__()
+        self.fn = fn
+        self.kwargs = kwargs
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        return self.fn(x, **self.kwargs)
+
+
+def apply_args_to_kwargs(
+    args: Sequence[Any], kwargs: Dict[str, Any], indexed_args: List[Tuple[int, str]]
+) -> Dict[str, Any]:
+    """
+    Applies the given arguments and keyword arguments to a dictionary of keyword arguments.
+
+    :param args: The positional arguments.
+    :param kwargs: The keyword arguments.
+    :param indexed_args: The list of pairs of indexes and names.
+
+    :return: A dictionary of keyword arguments with the applied arguments and keyword arguments.
+    """
+    args_dict: Dict[str, Any] = dict()
+    for idx, arg_name in indexed_args:
+        if idx < len(args):
+            args_dict[arg_name] = args[idx]
+        elif arg_name in kwargs:
+            args_dict[arg_name] = kwargs[arg_name]
+
+    return args_dict
+
+
+def extract_bn(model: nn.Module, graph: PTNNCFGraph, node: NNCFNode) -> ExtractedFunc:
+    """
+    Extract batch_norm operation.
+    If source modules inhered from nn.BatchNorm1d, nn.BatchNorm2d, or nn.BatchNorm3d return torch BatchNorm module.
```

Comment:
```suggestion
    If source modules inhered from nn.BatchNorm1d, nn.BatchNorm2d, or nn.BatchNorm3d, return torch BatchNorm module.
```

Reply:
Good catch 👍 but problem is not in comma.
This line is copypast that should be removed, now batch norm module is not required